internal/storage/storageutil/custom_retry.go

```diff
@@ -0,0 +1,78 @@
+package storageutil
+
+import (
+	"io"
+	"net"
+	"net/url"
+
+	"google.golang.org/api/googleapi"
+)
+
+func ShouldRetry(err error) (b bool) {
+	// HTTP 50x errors.
+	if typed, ok := err.(*googleapi.Error); ok {
+		if typed.Code >= 500 && typed.Code < 600 {
+			b = true
+			return
+		}
+	}
+
+	// HTTP 429 errors (GCS uses these for rate limiting).
+	if typed, ok := err.(*googleapi.Error); ok {
+		if typed.Code == 429 {
+			b = true
+			return
+		}
+	}
+
+	// HTTP 401 errors - Invalid Credentials
```

Comment:
move this extra logic to end after retry_bucket.go logic is done

Reply:
Added this logic in the last after the go-client logic.